DOCAPI-7436: Enum type docs. EN review, RU translation. (#6630)
ClickHouse/ClickHouse: docs/en/data_types/enum.md

# Enum

> Enumerated type storing pairs in `'string' = integer` format.

Reviewer: this sounds confusing since real data is stored only as integers, I'd suggest rewriting it without using "store"

ClickHouse supports:

- 8-bit `Enum`. It can contain up to 256 values enumerated in the `[-128, 127]` range.
- 16-bit `Enum`. It can contain up to 65536 values enumerated in the `[-32768, 32767]` range.

> ClickHouse automatically chooses the type of `Enum` when data is inserted. You can also use `Enum8` or `Enum16` types to lock in the storage size.

Reviewer: * second sentence does not make sense now
* needs a link to LowCardinality(String) data type as an alternative more flexible approach

Author: LowCardinality is not described yet, but I added your comment to the task for the type description.

ENGINE = TinyLog
```

> Column `x` can only store values that are listed in the type definition: `'hello'` or `'world'`. If you try to save any other value, ClickHouse will generate an exception. ClickHouse automatically chooses 8-bit when enumerating this `Enum`.

Reviewer: "enumerating Enum" sounds weird

```suggestion
Column `x` can only store values that are listed in the type definition: `'hello'` or `'world'`. If you try to save any other value, ClickHouse will raise an exception. 8-bit size for this `Enum` is chosen automatically.
```
